supportingFunctions.R

```diff
@@ -0,0 +1,118 @@
+single_txt_convert <- function(file_name){
+  data <- read.table(file_name, sep = " ", header = TRUE)
+  new_file_name <- gsub('txt', 'csv', file_name)
+  write.csv(data, file = new_file_name, row.names=F)
+}
+#then use the function above to define the function for converting multiple files
+txt_convert <- function(file_path){
+  file_name_list <- list.files(path = file_path, pattern = ".txt")
+  for(file in file_name_list){
+    name_including_path <- paste(file_path, file, sep = "/")
+    single_txt_convert(name_including_path)
+  }
+}
+
+#Compilation function
+#This function compiles all data of a given list of countries. Users should input a vector containing the name(s) of one or multiple countries.
+#usage: all_data <- compile(c("X","Y"), NA_remove = FALSE, NA_warning = FALSE)
+#requirements: the folders containing data from different countries should be put in the working directory, and they should be named like "countryX"
+#define a function to add columns of country and date to a dataframe
+add_col <- function(dataframe, country, date){
+  country_col <- rep(country, times = nrow(dataframe))
+  date_col <- rep(date, times = nrow(dataframe))
+  country_col <- data.frame(country_col)
+  date_col <- data.frame(date_col)
+  data_to_add <- cbind(country_col, date_col)
+  colnames(data_to_add) <- c("country","dayofYear")
+  dataframe <- cbind(dataframe, data_to_add)
+  return(dataframe)
+}
+#define a function to compile files of a single country
+compile_single_country <- function(country_name){
+  #create a void dataframe
+  single_country_data <- data.frame(matrix(ncol = 14, nrow = 0))
+  colnames(single_country_data) <- c("gender",	"age", "marker01", "marker02", 
+                                     "marker03", "marker04", "marker05", "marker06", 
+                                     "marker07", "marker08", "marker09", "marker10", 
+                                     "country", "dayofYear")
+  #get file names
+  file_path <- paste("./country", country_name, sep = "")
+  file_name_list <- list.files(path = file_path, pattern = ".csv")
+  #compile files
+  for(file in file_name_list){
+    file_name_including_path <- paste(file_path, file, sep = "/")
+    single_file_data <- read.csv(file_name_including_path, sep = ",", header = TRUE)
+    date <- substr(file, 8, 10)
+    single_file_data <- add_col(single_file_data, country_name, date)
+    single_country_data <- rbind(single_country_data, single_file_data)
+  }
+  return(single_country_data)
+}
+#define a function to compile all files
+compile <- function(country_name_list, NA_remove = FALSE, NA_warning = FALSE){
+  #create a void dataframe
+  all_data <- data.frame(matrix(ncol = 14, nrow = 0))
+  colnames(all_data) <- c("gender",	"age", "marker01", "marker02", 
+                                     "marker03", "marker04", "marker05", "marker06", 
+                                     "marker07", "marker08", "marker09", "marker10", 

```

Comment:
you can use paste("marker0", 1:10, sep = "") instead of typing all the markers' name